### PR TITLE
feat(monitors): Create missed checkins for monitor envs

### LINKED ISSUE #46137: Crons: Generate monitor errors in the correct environment

### PR DESCRIPTION
**Pull request by rjo100**
Properly updates MonitorEnvs for missed + failed checkins. Also moves analytics + events over to the MonitorEnvironment object

Closes https://github.com/getsentry/sentry/issues/46137

### CODE REVIEW

File: src/sentry/monitors/models.py [hunk 1]
```diff
@@ -55,18 +55,19 @@ def get_next_schedule(last_checkin, schedule_type, schedule):
...
-        "id": str(monitor.guid),
-        "slug": monitor.slug,
-        "name": monitor.name,
-        "config": monitor.config,
-        "status": monitor.get_status_display(),
-        "type": monitor.get_type_display(),
+        "id": str(monitor_environment.monitor.guid),
+        "slug": monitor_environment.monitor.slug,
+        "name": monitor_environment.monitor.name,
+        "environment": monitor_environment.environment.name,
```
Comment (rjo100) [inline src/sentry/monitors/models.py:67]: do we want to include `environment` in the context here? or just in the tags?

File: src/sentry/monitors/models.py [hunk 2]
```diff
@@ -400,6 +388,22 @@ def mark_failed(self, last_checkin=None, reason=MonitorFailure.UNKNOWN):
...
             return False
 
+        event_manager = EventManager(
+            {
+                "logentry": {"message": f"Monitor failure: {self.monitor.name} ({reason})"},
+                "contexts": {"monitor": get_monitor_environment_context(self)},
+                "fingerprint": ["monitor", str(self.monitor.guid), reason],
+                "tags": {
+                    "monitor.id": str(self.monitor.guid),
+                    "environment": self.environment.name,
```
Comment (rjo100) [inline src/sentry/monitors/models.py:398]: do I just send it as a tag? or as a top level param?

Comment (rjo100) [reply src/sentry/monitors/models.py:398]: I think it's a top level param, will change

File: src/sentry/monitors/models.py (hunk 1 above)
Comment (evanpurkhiser) [reply src/sentry/monitors/models.py:67]: It should just be in the event

File: src/sentry/monitors/tasks.py [hunk 3]
```diff
@@ -38,8 +38,8 @@ def check_monitors(current_datetime=None):
     if current_datetime is None:
         current_datetime = timezone.now()
 
-    qs = Monitor.objects.filter(
-        type__in=[MonitorType.CRON_JOB], next_checkin__lt=current_datetime
+    qs = MonitorEnvironment.objects.filter(
+        monitor__type__in=[MonitorType.CRON_JOB], next_checkin__lt=current_datetime
     ).exclude(
         status__in=[
```
Comment (evanpurkhiser) [inline src/sentry/monitors/tasks.py:44]: I think we need to exclude monitors that are disabled, we don't disable monitor environments yet i think right?

```suggestion
        monitor__status__in=[
```

Comment (rjo100) [reply src/sentry/monitors/tasks.py:44]: ah right. will just chain it instead of `Q`

**Review by evanpurkhiser — APPROVED** [review]
This looks good, let's get it in

**Pull request merged**